contracts/examples/DXswapRelayer.sol

```diff
@@ -0,0 +1,392 @@
+            orders[orderIndex].executed = true;
+            _pool(tokenA, tokenB, amountA, amountB, orders[orderIndex].amountA, orders[orderIndex].amountA);
+            emit ExecutedOrder(orderIndex);
+        } else {
+            /* Create an oracle to calculate average price before providing liquidity */
+            uint256 windowTime = _consultOracleParameters(amountA, amountB, reserveA, reserveB, maxWindowTime);
+            orders[orderIndex].oracleId = oracleCreator.createOracle(windowTime, pair);
+        }
+    }
+
+    function orderLiquidityRemoval(
+        address tokenA,
+        address tokenB,
+        uint256 liquidity,
+        uint256 amountA,
+        uint256 amountB,
+        uint256 priceTolerance,
+        uint256 minReserveA,
+        uint256 minReserveB,
+        uint256 maxWindowTime,
+        uint256 deadline,
+        address factory
+    ) external returns (uint256 orderIndex) {
+        require(factory == dxSwapFactory || factory == uniswapFactory, 'DXswapRelayer: INVALID_FACTORY');
+        require(msg.sender == owner, 'DXswapRelayer: CALLER_NOT_OWNER');
+        require(tokenA != tokenB, 'DXswapRelayer: INVALID_PAIR');
+        require(tokenA < tokenB, 'DXswapRelayer: INVALID_TOKEN_ORDER');
+        require(amountA > 0 && amountB > 0 && liquidity > 0, 'DXswapRelayer: INVALID_LIQUIDITY_AMOUNT');
+        require(priceTolerance <= PARTS_PER_MILLION, 'DXswapRelayer: INVALID_TOLERANCE');
+        require(block.timestamp <= deadline, 'DXswapRelayer: DEADLINE_REACHED');
+        require(maxWindowTime > 0, 'DXswapRelayer: INVALID_WINDOWTIME');
+
+        address pair = _pair(tokenA, tokenB, factory);
+        orderIndex = _OrderIndex();
+        orders[orderIndex] = Order({
+            action: REMOVAL,
+            tokenA: tokenA,
+            tokenB: tokenB,
+            amountA: amountA,
+            amountB: amountB,
+            liquidity: liquidity,
+            priceTolerance: priceTolerance,
+            minReserveA: minReserveA,
+            minReserveB: minReserveB,
+            oraclePair: pair,
+            deadline: deadline,
+            maxWindowTime: maxWindowTime,
+            oracleId: 0,
+            factory: factory,
+            executed: false
+        });
+
+        tokenA = tokenA == address(0) ? WETH : tokenA;
+        address dxSwapPair = DXswapLibrary.pairFor(address(dxSwapFactory), tokenA, tokenB);
+        (uint reserveA, uint reserveB,) = IDXswapPair(dxSwapPair).getReserves();
+        uint256 windowTime = _consultOracleParameters(amountA, amountB, reserveA, reserveB, maxWindowTime);
+        orders[orderIndex].oracleId = oracleCreator.createOracle(windowTime, pair);
+        emit NewOrder(orderIndex, REMOVAL);
+    }
+
+    function executeOrder(uint256 orderIndex) external {
+        Order storage order = orders[orderIndex];
+        require(orderIndex <= orderCount, 'DXswapRelayer: INVALID_ORDER');
```

Comment:
I missed this earlier, but this should be
```
require(orderIndex < orderCount);
```

i.e. 1 is not a valid order index when `orderCount` is 1